util/util.go

```diff
@@ -69,6 +69,22 @@ func RandomString(n int) string {
 	return string(result)
 }
 
+// GetMachinesForCluster returns a list of machines associated with the cluster.
```

Comment:
The `getMachines` method in `kubeadm_control_plane.controller.go` is very similar to this one. I wonder if we can move them both to the util package given that they might be useful elsewhere as well?

Reply:
Good point.

The result of `getMachines` is equivalent to the result of `GetMachinesForCluster`, filtered on the controlplane label, which is [defined in v1alpha3](https://github.com/kubernetes-sigs/cluster-api/blob/b4b7a1d0d266ca26febd1d113ca4786d664b1b9b/api/v1alpha3/machine_types.go#L30). Because the label is not defined by the kubeadm controlplane controller, I agree we can move this to the util package.

How about `util.GetControlPlaneMachinesForCluster`?

Reply:
Looks like [`util.GetControlPlaneMachines`](https://github.com/kubernetes-sigs/cluster-api/blob/4284af7625d8e45986fcca2bfe0ef3be60f88ecf/util/util.go#L104) exists already. You wrote about six months back :smile:. The function is a filter and it operates on a slice of pointers to Machines.

I would like the util package to define a standard way of working with CAPI types, so I want to avoid putting similar-but-slightly-different functions there if I can.

How about I open up an issue to
(a) standardize on how we pass around machines, i.e., with `[]*clusterv1.Machine` or `clusterv1.MachineList`.
(b) consolidate code from the `controlplane` package (and others) into the `util` package.

Reply:
Issue sounds good!

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/2062